### PR TITLE
Allow alignments to have clipping operations in the CIGAR

### PR DESCRIPTION
**Pull request by EricKutschera**
* Add new parameter: --allow-clipping
* Remove soft and hard clipping info from ends of CIGAR if allowed
* Add READ_CLIPPED as a read outcome for reads ignored due to clipping

Based on this issue: https://github.com/Xinglab/rmats-turbo/issues/39

### CODE REVIEW

**Review by ywang1993 — COMMENTED** [review]
In the `doSTARMapping` function, if s1 and s2 were provided, the alignment were done by `--alignEndsType EndToEnd`, which disable the soft clipping. I was wondering whether we should turn on the soft clipping for STAR if --allow_clipping is provided.

**Comment by EricKutschera** [comment]
That makes sense to me. I updated `doSTARMapping` so that `--alignEndsType EndToEnd` is not used when --allow_clipping is provided

**Review by ywang1993 — APPROVED** [review]

File: rMATS_pipeline/rmatspipeline/rmatspipeline.pyx [hunk 1]
```diff
@@ -303,21 +306,21 @@ cdef int is_bam_multijunc(const BamAlignment& bread,
...
                 return READ_NOT_EXPECTED_CIGAR
         else:
-            if bread.CigarData[i].Type != 'M':
+            if cigar_data[i].Type != 'M':
                 return READ_NOT_EXPECTED_CIGAR
 
-            length += bread.CigarData[i].Length
+            length += cigar_data[i].Length
 
     if (not variable_read_length) and length != readLength:
```
Comment (ywang1993) [inline rMATS_pipeline/rmatspipeline/rmatspipeline.pyx:325]: Same here, the comparison with input readLength should be done before dropping the soft/hard clipping BamAlignment data.

Comment (EricKutschera) [reply rMATS_pipeline/rmatspipeline/rmatspipeline.pyx:331]: The latest commit includes the clipped portion when comparing with readLength

**Review by ywang1993 — APPROVED** [review]

**Pull request merged**